ecc/bls12-377/multiexp.go

```diff
@@ -75,7 +76,7 @@ func (p *G1Jac) MultiExp(points []G1Affine, scalars []fr.Element, config ecc.Mul
 
 	// if nbTasks is not set, use all available CPUs
 	if config.NbTasks <= 0 {
-		config.NbTasks = runtime.NumCPU()
+		config.NbTasks = runtime.NumCPU() * 3
```

Comment:
yes, ok, should comment that part more.
essentially it's to avoid thread starvation; i.e in most cases it's better to have overall a bit more work with smaller tasks, such that usage of CPU core is 100% till the end (versus, say at the end only 3 tasks are running and CPU usage drops to 50% or something).

however this default value has some trade-offs; a machine can have N physical cores, M (N*2) threads (hyper-threading), but the optimal choice of c (window decomposition) is independent of that.. 

fwiw, *3 is good for M1 mbp, for the hpc6a AWS instance (96cores, no hyperthreading: "runtime.NumCPU()" returns 96). But on my desktop (8 cores, "16" with hyper-threading) *2 or *1 behaves better for some sizes.